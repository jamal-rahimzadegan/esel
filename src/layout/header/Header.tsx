import React from 'react';
import { useRouter } from 'next/router';
import { useHitAction } from 'hooks';
import { ICON_PATH } from 'constant';
import { HyperLink, Image } from 'components/index';
import createAction, { actions } from 'store/actions';
import { BackBtn, HeaderContainer } from '../LayoutStyle';

function Header(): JSX.Element {
  const { asPath: currentRoute, back } = useRouter();
  const hitAction = useHitAction();
  const isOnHomePage = currentRoute === '/home';

  const openDrawerMenu = () => hitAction(createAction(actions.OPEN_DRAWER));

  return (
    <HeaderContainer isOnHomePage={isOnHomePage}>
      <Image
        onClick={openDrawerMenu}
        className="pointer"
        objectFit="contain"
        src={ICON_PATH + 'menu.svg'}
        alt="menu"
        height={28}
        width={28}
      />
      <HyperLink href="/home">
        <Image objectFit="contain" src={ICON_PATH + 'logocrop.svg'} alt="logo" height={35} width={130} />
      </HyperLink>
      {!isOnHomePage ? (
        <BackBtn onClick={back} src={ICON_PATH + 'back.svg'} alt="back" height={40} width={40} />
      ) : (
        <div className="mr-5 ml-4" />
      )}
    </HeaderContainer>
  );
}

export default React.memo(Header);
