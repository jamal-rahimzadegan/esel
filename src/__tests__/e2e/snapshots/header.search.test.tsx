import renderer from 'react-test-renderer';
import { ProviderForTest } from '../../setupTests';
import '../../mocks/use-router';
import Header from '../../../layout/header/Header';

describe('Header ', () => {
  // Snapshot test
  it('should render header', () => {
    const component = (
      <ProviderForTest>
        <Header />
      </ProviderForTest>
    );
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
