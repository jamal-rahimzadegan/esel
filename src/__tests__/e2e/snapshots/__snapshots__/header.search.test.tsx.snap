// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header  should render header 1`] = `
<header
  className="LayoutStyle__HeaderContainer-sc-1txtlos-1 fFBrzl"
>
  <img
    alt="menu"
    className="ImgStyle-sc-1pe7p4e-0 csKpzj pointer"
    height={28}
    loading="lazy"
    onClick={[Function]}
    onError={[Function]}
    radius={null}
    src="/static/icon/menu.svg"
    width={28}
  />
  <a
    className="HyperLinkStyle__A-sc-27jl42-0 etQtyC"
    href="/home"
    onClick={[Function]}
    onMouseEnter={[Function]}
    rel="noopener noreferrer"
    target="_self"
    title="/home"
  >
    <img
      alt="logo"
      className="ImgStyle-sc-1pe7p4e-0 csKpzj"
      height={35}
      loading="lazy"
      onError={[Function]}
      radius={null}
      src="/static/icon/logocrop.svg"
      width={130}
    />
  </a>
  <div
    className="mr-5 ml-4"
  />
</header>
`;
